code/modules/mob/living/carbon/xenomorph/Facehuggers.dm

```diff
@@ -219,35 +231,88 @@
 		if(!attached) //Didn't hit anything?
 			i = 5
 			for(M in loc)
-				if(!i) break
+				if(!i)
+					break
 				if(CanHug(M))
 					Attach(M)
 					break
 				i--
 
-/obj/item/clothing/mask/facehugger/proc/Attach(mob/living/M)
-	set waitfor = 0
+/obj/item/clothing/mask/facehugger/throw_at(atom/target, range, speed)
+	. = ..()
+	if(stat == CONSCIOUS)
+		update_stat(UNCONSCIOUS) //stopping their process for the flight duration.
 
-	if(attached || M.status_flags & XENO_HOST || isXeno(M) || loc == M || stat != CONSCIOUS)
-		return
+/obj/item/clothing/mask/facehugger/throw_impact(atom/hit_atom, speed)
+	if(ismob(hit_atom) && stat != DEAD)
+		if(leaping && CanHug(hit_atom)) //Standard leaping behaviour, not attributable to being _thrown_ such as by a Carrier.
+			stat = CONSCIOUS
+			Attach(hit_atom)
+		else if(hit_atom.density) //We hit something, cool.
+			step(src, turn(dir, 180)) //We want the hugger to bounce off if it hits a mob.
+			throwing = FALSE
+			leaping = FALSE
+			addtimer(CALLBACK(src, .proc/fast_activate), 1.5 SECONDS)
+
+	else
+		addtimer(CALLBACK(src, .proc/GoActive), rand(MIN_ACTIVE_TIME,MAX_ACTIVE_TIME))
```

Comment:
```suggestion
		addtimer(CALLBACK(src, .proc/fast_activate), rand(MIN_ACTIVE_TIME,MAX_ACTIVE_TIME))
```
After that long time it might want to look around its surroundings.